src/services/item/plugins/tag/ItemTag.repository.ts

```diff
@@ -24,6 +25,38 @@ export class ItemTagRepository extends AbstractRepository<ItemTag> {
     return itemTags.map(({ tag }) => tag);
   }
 
+  async getCountBy({
+    search,
+    category,
+  }: {
+    search: string;
+    category?: TagCategory;
+  }): Promise<TagCount[]> {
+    if (!search) {
+      throw new IllegalArgumentException(`search is invalid: "${search}"`);
+    }
+
+    const q = this.repository
+      .createQueryBuilder('itemTag')
+      .select(['t.name AS name', 't.category AS category', 'count(t.id) as count'])
+      .leftJoinAndSelect('tag', 't', 't.id = itemTag.tag_id AND t.name ILIKE :search', {
+        search: `%${search}%`,
+      });
+
+    if (category) {
+      q.where('category = :category', { category });
+    }
+
+    const res = await q.groupBy('t.id').orderBy('count', 'DESC').getRawMany<{
+      name: Tag['name'];
+      category: Tag['category'];
+      count: string; // sql returns string by default
+    }>();
+
+    // transform into integer
+    return res.map((value) => ({ ...value, count: parseInt(value.count) }));
```

Comment:
If my suggestion below works correctly, you don't need to transform in the code.

```suggestion
    return res;
```